### PR TITLE
Improve Bramble extension loading via URL params, add extra/ dir

### PR DESCRIPTION
**Pull request by humphd**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,48 @@ Run one of the suggested local servers (or your own) from the root directory of
 
 Assuming you have Bramble running on port `8080`. Now you can visit [http://localhost:8080/src](http://localhost:8080/src).
 
+# Optional Extension Loading
+
+Bramble supports enabling and disabling various extensions via the URL and query params.
+A standard set of default extensions are always turned on:
+
+* CSSCodeHints
```
Comment (sedge) [inline README.md]: Could it be useful to link to each extension's documentation?

File: src/extensions/extra/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+List of extra extensions that can be optionally loaded into Bramble.
+If you add more extensions here, make sure you also update src/utils/BrambleExtensionLoader.js
+to reflect the new directories.  Also consider updating the top-level README.md.
```
Comment (sedge) [inline src/extensions/extra/README.md]: Newline

File: src/utils/BrambleExtensionLoader.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+        return extras;
+    }
+
+    exports.getExtensionList = function(params) {
+        return []
+            .concat(_processDefaults(params.get("disableExtensions")))
+            .concat(_processExtras(params.get("enableExtensions")));
+    };
+});
```
Comment (sedge) [inline src/utils/BrambleExtensionLoader.js]: This is a great refactor.

**Comment by sedge** [comment]
Travis needs to be fixed by adding these new modules to this section of the gruntfile: https://github.com/humphd/brackets/blob/bramble/Gruntfile.js#L362-L365

File: README.md (hunk 1 above)
Comment (humphd) [inline README.md]: No thanks :)

**Comment by humphd** [comment]
@sedge can you re-review?  I've fixed some other stuff in the grunt build process, too.  Tested locally in src/ and dist/ and all is well.  You can test with a url like `?enableExtensions=CodeFolding,brackets-cdn-suggestions,JSLint&disableExtensions=HTMLCodeHints,UrlCodeHints`

**Comment by sedge** [comment]
r+

**Comment by JordanTheriault** [comment]
r+, very good patch

**Pull request merged**